packages/io-ts-extra/src/__tests__/mapper.test.ts

```diff
@@ -51,6 +51,35 @@ describe('mapper', () => {
     expect(BoolToStringArray.encode(['f', 'a', 'l', 's', 'e'])).toEqual(false)
   })
 
+  it('map decodes nested mapped types', () => {
+    const Bool_Number = mapper(
+      t.boolean,
+      t.number,
+      b => (b ? 1 : 0),
+      n => n === 1
+    )
+
+    const Bool_Number_String = mapper(t.string, Bool_Number, JSON.parse, JSON.stringify)
```

Comment:
The naming here is confusing. For consistency this should be called `String_Bool_Number`